File: lab4 mpp/TeledonServices/ITeledonServices.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using teledon.domain; 

namespace teledon.services
{
    public interface ITeledonServices
    {
        void login(Voluntar voluntar, ITeledonObserver client);
        void logout(Voluntar voluntar, ITeledonObserver client);

        CazCaritabil[] GetCazuri();
        Donator[] GetDonatori();
    }
}

File: GUImpp/TeledonServices/src/main/java/teledon/services/ITeledonServices.java
package teledon.services;

import teledon.model.CazCaritabil;
import teledon.model.Donatie;
import teledon.model.Donator;
import teledon.model.Voluntar;

public interface ITeledonServices {

    void login(Voluntar voluntar, ITeledonObserver client) throws TeledonException;
    void logout(Voluntar voluntar, ITeledonObserver client) throws TeledonException;
    public Iterable<Donator> getAllDonatori() throws TeledonException;
    public void newDonation(Donatie donatie) throws TeledonException;
    public Iterable<CazCaritabil> getAllCazuri() throws TeledonException;


}
